docs/usage/generated-api.md

```diff
@@ -0,0 +1,199 @@
+}
+`
+</script>
+
+# Using Your Generated Code
+
+After [generating code](./generation.md), you will gain access to Zap's generated API. While this API is fully typesafe, meaning it will not throw errors for invalid usage, your static analysis tooling will. To solve this:
+
+- If you're using [VSCode](https://code.visualstudio.com/), we recommend installing the [Luau-LSP](https://marketplace.visualstudio.com/items?itemName=JohnnyMorganz.luau-lsp) extension.
+- If you're using Roblox Studio, you already have the proper Luau tooling.
+
+::: info
+This page will assume you're using the default casing value of `PascalCase`.
+
+Learn more about casing options [here](../config/options.md#casing).
+:::
+
+Zap generates two output files: one with the client API and one with the server API.
+
+## Event API
+
+In this section, we use the following Zap file as an example.
+
+<CodeBlock :code="eventConfigFile" />
+
+### Listening to Events
+
+Listening to events works the same on both the server and client. It is assumed that `Zap` is a properly defined reference to the generated API.
+
+If your event's [call field](../config/events.md#call) is `SingleAsync` or `SingleSync` you can assign only one listener the `SetCallback` function.
+
+```lua
+-- only server listeners are given the player argument
+Zap.AnotherEvent.SetCallback(function(Player, Foo, Bar)
+    -- Do something with the player and data
+end)
+```
+
+If your event's [call field](../config/events.md#call) is `ManyAsync` or `ManySync` you can assign multiple listeners using the `On` function.
+
+```lua
+local Disconnect = Zap.MyEvent.On(function(Options)
+    -- Do something with the data
+end)
+
+-- Disconnect the listener after 10 seconds
+task.delay(10, Disconnect)
+```
+
+As shown above, the `On` function for `Many` style events returns a `Disconnect` function, which can be used to remove the listener.
+
+::: danger
+Remember that synchronous event callbacks must not yield or error:
+
+- If a sync callback yields, it will cause undefined and game-breaking behavior.
+- If a sync callback errors, it will drop the packet.
+
+Use `Sync` events only when performance is critical.
+:::
+
+## Client API
+
+The client has a single function for firing events, `Fire`, which takes the event's data as its arguments.
+
+```lua
+Zap.AnotherEvent.Fire(true, 32)
+```
+
+## Server API
+
+The server has many functions for firing events, each with their own use case.
+
+::: tip
+`FireAll`, `FireExcept`, `FireList`, and `FireSet` serialize the event's data only once, making them more efficient than firing the event individually to each player.
+
+Use these functions when sending the same data to multiple players.
+:::
+
+### Fire
+
+The `Fire` function takes a player and the event's data as its arguments.
+
+```lua
+Zap.MyEvent.Fire(Player, {
+    foo = "baz",
+    bar = 1,
+})
+```
+
+### FireAll
+
+The `FireAll` function takes the event's data as its arguments. It will fire the event to all players.
```

Comment:
Maybe we should note this function can be disabled? See https://zap.redblox.dev/config/options.html#disable-fire-all for more info.